sklearn/datasets/tests/test_base.py

```diff
@@ -180,6 +181,16 @@ def test_load_iris():
     assert_equal(res.target_names.size, 3)
     assert_true(res.DESCR)
 
+    # test return_X_y option
+    X_y_tuple = load_iris(return_X_y=True)
+    bunch = load_iris()
+    assert_true(isinstance(X_y_tuple, tuple))
+    assert_equal(X_y_tuple[0].shape, bunch.data.shape)
```

Comment:
For large datasets, how about doing the extraordinarily naive thing of

1. Checking whether the shape of data and targets match up with the `X`, `y`
2. Check if the first sample of the bunched data and `X[0, :]` match up
3. Check if the last sample of the bunched data and `X[-1, :]` match up

?